lib/rules/prefer-const.js

```diff
@@ -96,6 +97,47 @@ function getDestructuringHost(reference) {
     return node;
 }
 
+/**
+ * Determines if a destructuring assignment node contains
+ * any MemberExpression nodes. This is used to determine if a
+ * variable that is only written once using destructuring can be
+ * safely converted into a const declaration.
+ * @param {ASTNode} node The ObjectPattern or ArrayPattern node to check.
+ * @returns {boolean} True if the destructuring pattern contains
+ *      a MemberExpression, false if not.
+ */
+function hasMemberExpressionAssignment(node) {
+    switch (node.type) {
+        case "ObjectPattern":
+            return node.properties.some(prop => {
+                if (prop) {
```

Comment:
Will this work correctly if `node` contains an object rest pattern?

```js
var { ...foo } = bar;
```

It seems like `prop.value` would be undefined in that case, which would cause the recursive call to crash.

Reply:
Fixed.